src/components/DatePicker/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import { useState } from 'react'
+import styled from '@emotion/styled'
+import { Text } from '../Text'
+import { Calendar } from '../../icons'
+import { space, radius, color } from '../../theme'
+import CalendarDialog from './CalendarDialog'
+import { TimeFrame } from '../../utils/dates'
+
+interface ButtonTextProps {
+  hasDate: boolean
+}
+
+interface DatePickerProps {
+  date?: Date | null
+  placeholder: string
+  title?: string
+  timeFrame?: TimeFrame
+  info?: string
+  monthLabel: string
+  yearLabel: string
+  locale?: string
+  onChange: (date: Date) => void
+}
+
+const StyledButton = styled.button`
+  padding: ${space[16]};
+  border-radius: ${radius.md};
+  background-color: ${color.stardust};
+  color: ${color.spaceLight};
+  font-family: inherit;
+  display: flex;
+  align-items: center;
+  margin-bottom: ${space[8]};
+  width: 100%;
+`
+
+const ButtonText = styled(Text)<ButtonTextProps>`
+  margin-left: ${space[16]};
+  color: ${props => (props.hasDate ? color.space : color.spaceLight)};
+`
+
+const DatePicker = ({
+  date = null,
+  placeholder,
+  title,
+  timeFrame = TimeFrame.future,
+  info,
+  monthLabel,
+  yearLabel,
+  locale = 'default',
+  onChange,
+}: DatePickerProps) => {
+  const [isOpen, setOpen] = useState(false)
+
+  const close = () => setOpen(false)
+  const toggle = () => setOpen(open => !open)
+
+  return (
+    <>
+      <CalendarDialog
+        date={date}
+        title={title}
+        timeFrame={timeFrame}
+        info={info}
+        monthLabel={monthLabel}
+        yearLabel={yearLabel}
+        locale={locale}
+        isOpen={isOpen}
+        close={close}
+        onChange={(date: Date) => {
+          onChange(date)
+          close()
+        }}
+      />
+      <StyledButton onClick={toggle}>
+        <Calendar size={24} color={color.spaceMedium} />
+        <ButtonText hasDate={!!date}>
+          {!!date
+            ? `${date.getDate()} ${date.toLocaleString(locale, {
+                month: 'short',
+              })} ${date.getFullYear()}`
```

Comment:
This is not localized. I don't know the specifics but not all countries use `day, month, year` 😉. Also not when writing it in full like this

Reply:
Nice catch, that's very true! We can actually do:

```
new Date().toLocaleString(locale, {
  day: 'numeric',
  month: 'short',
  year: 'numeric',
})
```

`Dec 13, 2021`

However, it adds a comma, which is not according to our new guidelines. I will check with @Marruk what we can do without adding any libs here 😅 

Another solution I thought of is leaving the responsibility of displaying the value to the user of the component, allowing the usage of any lib outside, but I'm not sure if I like that yet 💭

Reply:
@diogoredin Why not use `new Intl.DateTimeFormat('en-US').format(date))`

[MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat)

Reply:
It's also an option, but it has the same issue as the `toLocaleString` as it adds chars that do not conform with our new guidelines 😢  I'm checking this with @Marruk.

<img width="785" alt="Screenshot 2021-12-14 at 16 30 38" src="https://user-images.githubusercontent.com/1096800/146028707-8090620a-8a24-48c1-819c-c5bfcf5a2211.png">
uk.